dropwizard-auth/src/main/java/io/dropwizard/auth/AuthDynamicFeature.java

```diff
@@ -47,23 +49,31 @@ public AuthDynamicFeature(Class<? extends ContainerRequestFilter> authFilterClas
     public void configure(ResourceInfo resourceInfo, FeatureContext context) {
         final AnnotatedMethod am = new AnnotatedMethod(resourceInfo.getResourceMethod());
         final Annotation[][] parameterAnnotations = am.getParameterAnnotations();
-        //@DenyAll shouldn't be attached to classes
+        final Class<?>[] parameterTypes = am.getParameterTypes();
+
+        // First, check for any @Auth annotations on the method.
+        for (int i = 0; i < parameterAnnotations.length; i++) {
+            for (final Annotation annotation : parameterAnnotations[i]) {
+                // Optional auth requires that a concrete AuthFilter be provided.
+                if (annotation instanceof Auth && parameterTypes[i].equals(Optional.class) && authFilter != null) {
```

Comment:
I typically prefer flatter conditional tree, but in this case it seems reasonable. 👍